### ISSUE TITLE
Application is not installed (debug) to Android Smartphone

### ISSUE THREAD

**Issue by CoderAbhinav** [CoderAbhinav/boltlauncher-app#2]
# Application is not installed to the Android Phone connected to Mac
- The application just shows the following,
```
Launching lib/main.dart on [Redmi Note 7S] in debug mode...
Running Gradle task 'assembleDebug'...
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app-debug.apk...
```
- It's not able to install the application & shows the following error
```
Error: ADB exited with exit code 1
Performing Streamed Install

adb: failed to install /Users/abhinavbelhekar/Documents/MOTORISED-BALLISTA-CONTROLLER-WORKSPACE/boltlauncher/build/app/outputs/flutter-apk/app-debug.apk:
Error launching application on Redmi Note 7S.
```

**Comment by CoderAbhinav** [comment]
- Tried factory resetting the smartphone (x Not working)
- Error persists

**Comment by CoderAbhinav** [comment]
- [Link to ADB issue tracker](https://issuetracker.google.com/issues/270205252?pli=1)
- Use following command before you start running the debug version
```
export ADB_LIBUSB=1
adb kill-server && adb root
```
- 
- Read the following quote from comment

"The commands you ran did two things:

export ADB_LIBUSB=1: This command sets the ADB_LIBUSB environment variable to 1. When this variable is set, ADB uses the libusb library instead of the default USB library. This can sometimes resolve issues with device detection if there are problems with the default USB library.

adb kill-server && adb root: This command does two things. First, adb kill-server stops the ADB server process. Then, adb root restarts the ADB server with root permissions. This can resolve issues with device detection if the ADB server was not running or was running without the necessary permissions.

So, in summary, your issue was likely due to a problem with ADB, either because the Android SDK was not installed correctly, the ANDROID_SDK_ROOT environment variable was not set correctly, or the ADB server was not running with the necessary permissions. The commands you ran changed how ADB interacts with USB devices and restarted the ADB server with root permissions, which resolved the issue."
